src/Consolonia.Core/Drawing/PixelBufferImplementation/Symbol.cs

```diff
@@ -0,0 +1,156 @@
+        public Symbol(char ch)
+        {
+            Text = SymbolCache[ch];
+            if (Text == null)
+            {
+                Text = ch.ToString();
+                SymbolCache[ch] = Text;
+            }
+
+            Width = (byte)Text.MeasureText();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Symbol(byte upRightDownLeftPattern)
+        {
+            Width = 1;
+            Pattern = upRightDownLeftPattern;
+            char boxChar = BoxPattern.GetBoxChar(upRightDownLeftPattern);
+            if (boxChar == BoxPattern.BoldChar)
+            {
+                // get well-known string instance
+                Text = BoldText;
+            }
+            else if (boxChar == BoxPattern.EmptyChar)
+            {
+                Text = string.Empty;
+            }
+            else if (boxChar >= BoxPattern.Min && boxChar <= BoxPattern.Max)
+            {
+                // get cached string instance
+                Text = SymbolCache[boxChar];
+                if (Text == null)
+                {
+                    Text = boxChar.ToString();
+                    SymbolCache[boxChar] = Text;
+                }
+            }
+            else
+                // uhoh
+            {
+                throw new ArgumentOutOfRangeException(nameof(upRightDownLeftPattern));
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Symbol(string glyph)
+        {
+            if (glyph.Length == 1)
+            {
+                Text = SymbolCache[glyph[0]];
+                if (Text == null)
+                {
+                    Text = glyph[0].ToString();
+                    SymbolCache[glyph[0]] = Text;
+                }
+            }
+            else
+            {
+                Text = glyph;
+            }
+
+            Width = (byte)Text.MeasureText();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Symbol(Rune rune)
+            : this(rune.ToString())
+        {
+        }
+
+        public bool Equals(Symbol other)
+        {
+            return Text.Equals(other.Text, StringComparison.Ordinal) &&
+                   Pattern == other.Pattern;
+        }
+
+        public string Text { get; }
+
+        // box pattern for box merging.
+        public byte Pattern { get; }
+
+        [JsonIgnore] public byte Width { get; init; }
+
+        public bool NothingToDraw()
+        {
+            return string.IsNullOrEmpty(Text);
+        }
+
+        public Symbol Blend(ref Symbol symbolAbove)
+        {
+            if (symbolAbove.NothingToDraw())
+                return this;
+
+            if (IsBoxSymbol() && symbolAbove.IsBoxSymbol())
+                return new Symbol(BoxPattern.Merge(Pattern, symbolAbove.Pattern));
+            return symbolAbove;
+        }
+
+        public bool IsBoxSymbol()
+        {
+            return Pattern > 0;
```

Comment:
So may be it should be inclusive

Reply:
I don't think so, if pattern is zero we aren't tracking any single or double lines. This makes sense to me and it's what we use for all of the chars that aren't sourced from box pattern.